src/main/scala/uncore/tilelink2/Crossing.scala

```diff
@@ -15,16 +15,24 @@ class TLAsyncCrossingSource(sync: Int = 3) extends LazyModule
     val io = new Bundle {
       val in  = node.bundleIn
       val out = node.bundleOut
+      val bypass = Bool(INPUT)
     }
 
     ((io.in zip io.out) zip (node.edgesIn zip node.edgesOut)) foreach { case ((in, out), (edgeIn, edgeOut)) =>
+      val reset_n = out.a.sink_reset_n
       val bce = edgeIn.manager.anySupportAcquire && edgeIn.client.anySupportProbe
       val depth = edgeOut.manager.depth
 
       out.a <> ToAsyncBundle(in.a, depth, sync)
       in.d <> FromAsyncBundle(out.d, sync)
 
+      assert (io.bypass || reset_n, "Unsafe to reset sink manager without first asserting source bypass")
+
       if (bce) {
+        val up = RegInit(Bool(false))
+        when (reset_n) { up := Bool(true) }
```

Comment:
You need to pass reset_n, because isolation brings the wires to zero. We want isolation to appear the same as reset to the other side.